### PR TITLE
migrate to null safety

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

File: lib/src/handler.dart [hunk 1]
```diff
@@ -83,15 +83,13 @@ class ShelfTestHandler {
...
-
-      throw TestFailure('$description handler returned null for $requestInfo.');
+      return await expectation.handler(request);
     } on HijackException catch (_) {
       rethrow;
     } catch (error, stackTrace) {
       _zone.handleUncaughtError(error, stackTrace);
-      return null;
+      // We can't return null here, the handler type doesn't allow it.
+      return Response.internalServerError(body: 'Response Hijacked');
```
Comment (jakemac53) [inline lib/src/handler.dart:92]: I was a bit unsure what to do here, we are not allowed to return null.

Comment (natebosch) [reply lib/src/handler.dart:92]: I don't think `'Response Hijacked'` would be the appropriate text if this is the pattern we use.

What would happen if we also `rethrow` here?

Comment (jakemac53) [reply lib/src/handler.dart:92]: I really have no idea what the intention of this code is. I don't see any existing tests for the hijack behavior so I can't tell what is expected...

Comment (jakemac53) [reply lib/src/handler.dart:92]: Oh wait I just realized that this wasn't even in the hijack block lol, so this is just catching general errors.

If I rethrow it makes the tests fail. Maybe just a `Response.notFound('')` would be appropriate?

Comment (jakemac53) [reply lib/src/handler.dart:92]: Ya this is intentionally forwarding the errors to the original zone where the ShelfTestHandler was created. Rethrowing makes it show up in the current zone as well.

I think just some sort of placeholder response is the correct thing here, even though its a bit weird.

Comment (jakemac53) [reply lib/src/handler.dart:92]: I went with `return Response.internalServerError(body: '$error');` as that does kind of make sense? I don't think this is super important actually.

Comment (natebosch) [reply lib/src/handler.dart]: `return Response.internalServerError(body: '$error');` SGTM.

**Review by natebosch — APPROVED** [review]

**Pull request merged**